### PR TITLE
Feature/chat demo

### PR DESCRIPTION
**Pull request by dpwspoon**
This is taken from https://github.com/aws/amazon-chime-sdk-component-library-react/tree/master/demo/chat which will be deprecated on move.
It adds support for authentication via STS lambda.

**Issue #:**
n/a

**Description of changes:**
There are 2 parts to this Pull request:
1) This is part of a consolidation of Chime demos into a single repository
2) This adds an alternative way to authenticate.  The previous demo showed Cognito User Pools.  This adds an alternative which is to run a small backed service (in this case via API gateway and Lambda) that grants AWS credentials.  

**Testing**

1. How did you test these changes?

Ran through README on how to test for both Cognito and Credential Service Flow

2. Can these changes be tested using one of the demo application? If yes, which demo application can be used to test it?

Yes, this change is a demo...

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/chat/README.md [hunk 1]
```diff
@@ -31,11 +31,11 @@ This doc is intended for developers interested in the Amazon Chime Chat SDK, eve
 ## Running the Amazon Chime Sample App
 
 1. Ensure your workspace has node.js installed. Type `node -v` in your terminal to confirm, and it should return a version number.
-2. Navigate to the root folder of Amazon Chime Sample App `demo/chat`
-3. Open `src/Config.js` with the editor of your choice and update the each missing config with the values from the
+2. Navigate to the root folder of Amazon Chime Sample App `apps/chat`
+3. In the root directory `apps/chat`, Open `src/Config.js` with the editor of your choice and update the each missing config with the values from the
```
Comment (ltrung) [inline apps/chat/README.md:35]: nit: open

File: apps/chat/README.md [hunk 2]
```diff
@@ -73,9 +73,10 @@ Skip ahead to [Creating a Channel](#creating-a-channel)
 1. Open a browser of your choice and navigate to [http://localhost:9000](http://localhost:9000/) to access the client
 2. Change the drop down to Credential Exchange Service
 3. The Credential Exchange Service is a small lambda running behind API gateway that enables exchanging your applications or identity 
-provider's (IDP) token for AWS credentials.  To simulate the processes of exchanging credentials by default the lambda returns anonymous access
-regardless of the token provided.  Click "Exchange Token for AWS Credentials" to get anonymous access to the chat application.  If you
-wish to change the code to validate your application/IDP token, modify the following code in /backend/serverless/template.yml.
+provider's (IDP) token for AWS credentials, or for you to implement custom authentication.  To simulate the processes of exchanging
```
Comment (ltrung) [inline apps/chat/README.md:76]: nit: Double space.

File: apps/chat/README.md [hunk 3]
```diff
@@ -73,9 +73,10 @@ Skip ahead to [Creating a Channel](#creating-a-channel)
 1. Open a browser of your choice and navigate to [http://localhost:9000](http://localhost:9000/) to access the client
```
Comment (ltrung) [inline apps/chat/README.md:73]: nit: . at end of sentence?

**Review by ltrung — COMMENTED** [review]
Doesn't seem ` apps/chat/src/components/ChannelModals/AddModeratorsChannelModal.jsx` (blank file) is used? Can we remove it?

**Review by ltrung — COMMENTED** [review]
Need to include Config file in .gitignore to prevent commit values from Config.

**Comment by dpwspoon** [comment]
All feedback applied, thanks

File: apps/chat/README.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+4. In the root directory `apps/chat`, run `npm install` and then `npm start` to build and start the client
+5. Open https://localhost:9000/ in your browser
+6. By default the demo uses Amazon Cognito User Pools to manage users and login.  Alternatively, you can get credentials through a small 
+ lambda service that the cloudformation template sets up.  If you prefer to use the default Cognito flow
+ continue with [Cognito User Pools - Register a New User](#cognito-user-pools-register-a-new-user).  If you prefer to use the credential 
+ exchange service see [Credential Exchange Service: Login](#Credential-Exchange-Service-Login).
+
+#### Cognito User Pools: Register a New User
+
+New users can register through the Amazon Chime Sample App.+ (nullable NSString *)serializeTimestamp:(NSDictionary *)rules value:(id)value error:(NSError *__autoreleasing *)error {
```
Comment (ltrung) [inline apps/chat/README.md:46]: This doesn't seem to be render correctly for me (The code shows in the right hand side)

**Review by ltrung — APPROVED** [review]

**Pull request merged**